### ISSUE TITLE
repo_deps_libyear current_release_date and latest_release_date data type

### ISSUE THREAD

**Issue by cdolfi** [chaoss/augur#2663]
current_release_date and latest_release_date is a string type not a date type, and the format is not convertible with pandas

**Comment by ChildEater69** [comment]
You can use these formats to convert a string '2022-01-01' into date type using  '%Y-%m-%d' as format specifier

# Convert string dates to datetime format
df['current_release_date'] = pd.to_datetime(df['current_release_date'], format='%Y-%m-%d')
df['latest_release_date'] = pd.to_datetime(df['latest_release_date'], format='%Y-%m-%d')

![Jupyter Run for above code](https://github.com/chaoss/augur/assets/115105709/ed91fcee-d22f-41a7-8779-ff4307c90613)

**Comment by ChildEater69** [comment]
you can use %Y-%m-%d is the format specifier for a date string like "2022-01-01"

# Convert string dates to datetime format

df['current_release_date'] = pd.to_datetime(df['current_release_date'], format='%Y-%m-%d')
df['latest_release_date'] = pd.to_datetime(df['latest_release_date'], format='%Y-%m-%d')


![Jupyter Run for above code](https://github.com/chaoss/augur/assets/115105709/752edcc0-d847-4c78-a50d-18e786a0210f)

**Comment by cdolfi** [comment]
@ChildEater69 The formatting of the columns are off thats why that conversion is not working. It is not of the %Y-%m-%d format.

**Comment by sgoggins** [comment]
@cdolfi : OK. Noted.

**Comment by MoralCode** [comment]
can confirm: https://github.com/chaoss/augur/blob/4221cba76e6ce93e8d4e2f2676f1525bfeacc8ca/augur/application/db/models/augur_data.py#L1981-L1982

**Comment by PredictiveManish** [comment]
can we store this in database as date

> can confirm:
> 
> [augur/augur/application/db/models/augur_data.py](https://github.com/chaoss/augur/blob/4221cba76e6ce93e8d4e2f2676f1525bfeacc8ca/augur/application/db/models/augur_data.py#L1981-L1982)
> 
> Lines 1981 to 1982 in [4221cba](/chaoss/augur/commit/4221cba76e6ce93e8d4e2f2676f1525bfeacc8ca)
> 
>  current_release_date = Column(String) 
>  latest_release_date = Column(String)

In the same file `current_version` is also `current_verion` and it's uniform all across whole code, should I touch that or not?

**Comment by PredictiveManish** [comment]
> can confirm:
> 
> [augur/augur/application/db/models/augur_data.py](https://github.com/chaoss/augur/blob/4221cba76e6ce93e8d4e2f2676f1525bfeacc8ca/augur/application/db/models/augur_data.py#L1981-L1982)
> 
> Lines 1981 to 1982 in [4221cba](/chaoss/augur/commit/4221cba76e6ce93e8d4e2f2676f1525bfeacc8ca)
> 
>  current_release_date = Column(String) 
>  latest_release_date = Column(String)

Only this one or changes in Database schema too? https://github.com/chaoss/augur/blob/7e07c1dbba6f206c1a2c8f567d08058191aa49c2/augur/application/schema/augur_full.sql#L2134-L2135

**Comment by MoralCode** [comment]
@cdolfi can you clarify what you meant by "and the format is not convertible with pandas"?

i.e. is it a common date format that pandas doesnt support easily? or does the format of the data vary so wildly that its hard to parse because these are string fields?

**Comment by cdolfi** [comment]
I am pretty sure the latter but it's been about 2 years since I created this so my memory is a bit fuzzy on it

**Comment by MoralCode** [comment]
Sean's queries in https://github.com/chaoss/augur/pull/3384#discussion_r2524174792 seem like they were able to convert all the values on his instance

**Comment by MoralCode** [comment]
seems like current discussion around this is mostly related to: whats the best way to do this migration that accounts for existing values people may have in their databases?
